lib/matplotlib/ticker.py

```diff
@@ -1214,12 +1206,43 @@ class EngFormatter(Formatter):
          24: "Y"
     }
 
-    def __init__(self, unit="", places=None):
+    def __init__(self, unit="", places=None, sep=" "):
+        """
+        Parameters
+        ----------
+        unit : str (default: "")
+            Unit symbol to use, suitable for use with single-letter
+            representations of powers of 1000. For example, 'Hz' or 'm'.
+
+        places : int (default: None)
+            Precision with which to display the number, specified in
+            digits after the decimal point (there will be between one
+            and three digits before the decimal point). If it is None,
+            the formatting falls back to the floating point format '%g',
+            which displays up to 6 *significant* digits (i.e. the
+            equivalent value for *places* varies between 0 and 5, both
```

Comment:
Also, now that I think of it, this behavior (which was already present before, I know) is a bit weird, isn't it?  `f` also defaults to 6 decimal places (not significant digits, I know) (see `pydoc FORMATTING`), so basically the possible formats are

.6g (== g) if places == None
.0f .1f .2f .3f etc. if places == 0 1 2 3 etc.

It would probably be a bit awkward to change this, but it looks like in an idea API places should be set by one parameter (perhaps defaulting to 6) and f vs g by another.